src/modules/LaptopDetail/components/CommentTab/LaptopComment.tsx

```diff
@@ -0,0 +1,179 @@
+import { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import {
+  Rate,
+  Typography,
+  Input,
+  Divider,
+  Pagination,
+  notification
+} from "antd";
+import * as Yup from "yup";
+import { Controller, useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+
+import { CommonButton } from "components";
+import { useAppDispatch, useAppSelector } from "hooks/useRedux";
+import { RootState } from "redux/store";
+import { IComment, ICommentInput } from "types/comment.model";
+import {
+  addNewCommentActionRequest,
+  setNullCommentItem,
+  updateCommentByIdActionRequest
+} from "redux/features/laptop.slice";
+import { IProfile } from "types/profile.model";
+import {
+  CommentList,
+  StarEvaluation,
+  SubmitCommentButton
+} from "./CommentTab.styled";
+import CommentCard from "./CommentCard";
+
+const { TextArea } = Input;
+const { Title } = Typography;
+
+const LaptopComment = () => {
+  const { t } = useTranslation();
+  const [pageNum, setPageNum] = useState<number>(1);
+  const dispatch = useAppDispatch();
+  const { commentList, commentItem, laptopDetail, loading } = useAppSelector(
+    (state: RootState) => state.laptop
+  );
+  const { userInfo, profile } = useAppSelector(
+    (state: RootState) => state.auth
+  );
+
+  const handleChangePageNum = (page: number) => {
+    setPageNum(page);
+  };
+
+  const {
+    handleSubmit,
+    control,
+    reset,
+    watch,
+    formState: { errors, isValid }
+  } = useForm<ICommentInput>({
+    mode: "onChange",
+    defaultValues: {
+      laptopId: laptopDetail?._id,
+      userId: userInfo?._id,
+      updatedAt: Date.now().toString(),
+      rating: 5,
+      comment: ""
+    },
+    resolver: yupResolver(
+      Yup.object().shape({
+        comment: Yup.string().required(`${t("auth:is_required")}`)
+      })
+    )
+  });
+
+  const handleSubmitCommentForm = (data: ICommentInput) => {
+    if (userInfo && profile) {
```

Comment:
change If/ else to a better version using if only

Reply:
I have just tested all circumtance. If i use if only instead of if/else it will do wrong requirement

Reply:
if (!userInfo || !profile) {
 return someLogic
 }
 
continue other logic

extract old block of code not to have many if/else statement on it

Reply:
oke thanks bro.